### PR TITLE
Use `is not null` check when available.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
part of https://github.com/dotnet/roslyn/issues/42368

Review a commit at a time.

### CODE REVIEW

File: src/Analyzers/CSharp/CodeFixes/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsCodeFixProvider.cs [hunk 1]
```diff
@@ -26,33 +29,46 @@ protected override string GetIsNullTitle()
...
-                .BinaryExpression(
-                    SyntaxKind.IsExpression,
-                    (ExpressionSyntax)argument,
-                    SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.ObjectKeyword)))
-                .Parenthesize();
+            var parseOptions = (CSharpParseOptions)argument.SyntaxTree.Options;
+
+            // TODO: this needs to be CSharp9 once that version is available.
+#if !CODE_STYLE
+            if (parseOptions.LanguageVersion >= LanguageVersion.CSharp8)
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/CodeFixes/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsCodeFixProvider.cs:50]: note: this is wrong.  discussing with @gafter getting an actual CSharp9 lang version that we can test against.

File: src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckForReferenceEqualsTests.cs [hunk 2]
```diff
@@ -17,7 +17,9 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseIsNullCheck
     public partial class UseIsNullCheckForReferenceEqualsTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
     {
         private static readonly ParseOptions CSharp7 = CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp7);
-        private static readonly ParseOptions CSharp9 = CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp8);
+
+        // Switch this to actually reference LanguageVersion.CSharp9 when available.
```
Comment (dibarbet) [inline src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckForReferenceEqualsTests.cs:21]: do we have a tracking issue for that availability?

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckForReferenceEqualsTests.cs:21]: There's an email thread going on right now about this.  I can open a tracking issue though to make sure htis doesn't get lost.

**Comment by CyrusNajmabadi** [comment]
@jcouv  https://github.com/dotnet/roslyn/pull/43322/commits/4af20fb2de19da701942ee76f1734abe21548a52  Thanks!

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/LanguageVersionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Shared.Extensions
+{
+    internal static class LanguageVersionExtensions
+    {
+        // https://github.com/dotnet/roslyn/issues/43348
+        //
+        // This method should be removed when LanguageVersion.CSharp9 is available, and all callers should just
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/LanguageVersionExtensions.cs:11]:
```suggestion
        // This value should be removed when LanguageVersion.CSharp9 is available, and all callers should just
```

File: src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckForReferenceEqualsTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckForReferenceEqualsTests.cs]: https://github.com/dotnet/roslyn/issues/43348.  Thanks!

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @jcouv @jaredpar  can you take a quick look?  The only changes in the compiler layer are just some comments explaining a process we need to update compiler/IDE once CSharp9 lang-version is available.

Thanks!

File: src/Analyzers/CSharp/CodeFixes/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsCodeFixProvider.cs [hunk 4]
```diff
@@ -6,13 +6,17 @@
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Shared.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.UseIsNullCheck;
 
 namespace Microsoft.CodeAnalysis.CSharp.UseIsNullCheck
 {
+    using static SyntaxFactory;
```
Comment (jaredpar) [inline src/Analyzers/CSharp/CodeFixes/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsCodeFixProvider.cs:15]: Generally all usings should be above the namespace. Don't think `using static` is treated differently for that rule.

Comment (CyrusNajmabadi) [reply src/Analyzers/CSharp/CodeFixes/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsCodeFixProvider.cs:15]: It is.  Manish is tracking this as this is so common we're going to have a specific option for it. :)

**Pull request merged**